test_data/asm/trait_parsing.asm

```diff
@@ -0,0 +1,16 @@
+mod types {
+    enum DoubleOpt<T> {
+        None,
+        Some(T, T)
+    }
+
+    trait ArraySum<T, DoubleOpt<T> > {
+        array_sum: T[4 + 1] -> DoubleOpt,
```

Comment:
this is what I meant:
```suggestion
    trait ArraySum<T> {
        array_sum: T[4 + 1] -> DoubleOpt<T>,
```

And if you want to be even more thorough, you could check that when this has been exported to pil (if there is a set of suitable tests for that), it says `types::DoubleOpt<T>`